lib/SIL/IR/ValueOwnership.cpp

```diff
@@ -212,6 +210,12 @@ CONSTANT_OR_NONE_OWNERSHIP_INST(Guaranteed, OpenExistentialBoxValue)
 // value).
 CONSTANT_OR_NONE_OWNERSHIP_INST(Owned, MarkUninitialized)
 
+// In raw SIL, a MoveValue delimits the scope of trivial variables.
+CONSTANT_OR_NONE_OWNERSHIP_INST(Owned, MoveValue)
+
+// In raw SIL, a BeginBorrow delimits the scope of trivial variables.
```

Comment:
Maybe we won't need to support `begin_borrow : $Trivial`.
Also, it might be better to introduce a new opcode for trivial `move_value`. I originally wanted as much consistency as possible between trivial/nontrivial vardecls in diagnostics. If we want perfect consistency though, the variable scope would use separate scope markers: `%token = begin_vardecl` -> `end_vardecl %token` instead of flags. `begin_vardecl` would only produce a token, so we don't need to add use-def handling, and those scoped can be dropped after diagnostics. But I don't know if it's realistic at this point to change all code that relies on move_value/begin_borrow.

Reply:
Maybe it would be best to postpone permitting `begin_borrow %t : $Trivial` until it actually gets used.

Reply:
I was convinced that we needed `begin_borrow` until after I rewrote `getValueForLexicalLifetimeBinding`